src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Entry<K, V>[] table;
+    private int size = 0;
+    private Entry<K, V> nullKeyEntry;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        table = (Entry<K, V>[]) new Entry[INITIAL_CAPACITY];
+    }
+    
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+        private Entry<K, V> next;
+
+        public Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            if (nullKeyEntry == null) {
+                nullKeyEntry = new Entry<>(null, value,null);
+                size++;
+            } else {
+                nullKeyEntry.value = value;
+            }
+            return;
+        }
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
 
+        for (Entry<K, V> e = table[index]; e != null; e = e.next) {
+            if (e.key.equals(key)) {
+                e.value = value;
+                return;
+            }
+        }
+
+        addEntry(hash, key, value,index);
```

Comment:
Using public access modifiers everywhere is not recommended. If the method 'addEntry' is used only inside the same class, it should not be `public`. Keep your code as close as possible to follow the encapsulation principle.